src/store/modules/auth.js

```diff
@@ -85,6 +96,43 @@ export const signupUser = userData => {
   };
 };
 
+export const socialAuth = (history, location) => {
+  return dispatch => {
+    try {
+      dispatch(loginInitialize());
+      let query = location.search;
```

Comment:
We can use `qs` to handle query string parsing.
https://www.npmjs.com/package/qs

Reply:
Done.